### PR TITLE
feat: support deployment-id

### PR DESCRIPTION
**Pull request by imLinguin**
In case of CYGNI it allows the game to work
obraz

key points of this feature
- sidecar config is stored in InstalledGame class
- the config is updated during the launch if revision changed without the game update
- EGL export will write the `.egstore/<appName>appconfig.json` file as well

### CODE REVIEW

File: legendary/core.py [hunk 1]
```diff
@@ -1860,6 +1873,18 @@ def egl_export(self, app_name):
...
+            # EGL seems to change keys to Pascal
+            sidecar_conf = lgd_igame.sidecar['config']
+            json_config = json.loads(sidecar_conf)
+            new_config = dict()
+            # Make config PascalCase
+            for key in json_config:
+                new_config[key.capitalize()] = json_config[key]
+
+            with open(os.path.join(egstore_folder, f'{lgd_game.app_name}appconfig.json', ), 'w') as ac:
+                ac.write(json.dumps(new_config))
```
Comment (CommandMC) [inline legendary/core.py]:
```suggestion
                json.dump(new_config, ac)
```

File: legendary/cli.py [hunk 2]
```diff
@@ -580,7 +580,8 @@ def launch_game(self, args, extra):
             return self._launch_origin(args)
 
         igame = self.core.get_installed_game(app_name)
-        if (not igame or not igame.executable) and (game := self.core.get_game(app_name)) is not None:
+        game = self.core.get_game(app_name)
+        if (not igame or not igame.executable) and game is not None:
```
Comment (CommandMC) [inline legendary/cli.py:584]: Why was this changed?

**Review by CommandMC — APPROVED** [review]
LGTM, only got a few tiny nitpicks

File: legendary/cli.py (hunk 2 above)
Comment (imLinguin) [reply legendary/cli.py:584]: so that game is accessible by get_cdn_urls call below

Comment (CommandMC) [reply legendary/cli.py:584]: Ah right, forgot that the operator worked like that. ...I should really write more Python again
